### PR TITLE
devDeps: @metamask/eslint-config* from v11 to v12

### PR DESCRIPTION
**Pull request by legobeat**
Was previously also part of #98

### CODE REVIEW

**Comment by legobeat** [comment]
@SocketSecurity ignore builtins@5.0.1
    @SocketSecurity ignore eslint-plugin-n@15.7.0

**Review by mcmire — DISMISSED** [review]
Looks good!

**Review by Mrtenz — DISMISSED** [review]

**Review by Mrtenz — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -46,27 +46,29 @@
...
-    "@metamask/eslint-config-nodejs": "^11.1.0",
-    "@metamask/eslint-config-typescript": "^11.1.0",
+    "@metamask/eslint-config": "^12.0.0",
+    "@metamask/eslint-config-jest": "^12.0.0",
+    "@metamask/eslint-config-nodejs": "^12.0.0",
+    "@metamask/eslint-config-typescript": "^12.0.0",
     "@types/jest": "^28.1.6",
-    "@typescript-eslint/eslint-plugin": "^5.42.1",
-    "@typescript-eslint/parser": "^5.42.1",
+    "@types/node": "^16",
```
Comment (Mrtenz) [inline package.json:54]: Why do we need this now?

**Review by Mrtenz — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (legobeat) [reply package.json:54]: This was in order to indicate the version to use by other devDependencies (which already pull in `@types/node@*`. Looks like this wasn't done properly, but is now: 54c4f41be2da592e56f83a877428cbdec02819e9

Arguably `devDependencies` isn't the field to use for this - open to dropping it.

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**